src/event.rs

```diff
@@ -502,48 +514,75 @@ where
 				);
 				let payment_preimage = match purpose {
 					PaymentPurpose::Bolt11InvoicePayment { payment_preimage, payment_secret } => {
-						if payment_preimage.is_some() {
-							payment_preimage
-						} else {
-							self.channel_manager
-								.get_payment_preimage(payment_hash, payment_secret)
-								.ok()
-						}
+						payment_preimage.or(self
+							.channel_manager
+							.get_payment_preimage(payment_hash, payment_secret)
+							.ok())
```

Comment:
Need to consider this further, but I thought a `None` preimage meant that the user needs to supply it, right? How could we get it from `ChannelManager`?